extensions/sockets/src/test/java/org/mule/extension/socket/TcpStreamingTestCase.java

```diff
@@ -0,0 +1,97 @@
+ * LICENSE.txt file.
+ */
+package org.mule.extension.socket;
+
+import static java.lang.String.format;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import org.mule.functional.functional.EventCallback;
+import org.mule.functional.functional.FunctionalStreamingTestComponent;
+import org.mule.runtime.core.api.MuleEventContext;
+
+import java.io.ByteArrayInputStream;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.Test;
+
+public class TcpStreamingTestCase extends SocketExtensionTestCase
+{
+
+    public static final String TEST_MESSAGE = "Test TCP Request";
+    public static final String RESULT = "Received stream; length: %d; 'Test...uest'";
+    private static int SINGLE_ITERATION = 1;
+
+    @Override
+    protected String getConfigFile()
+    {
+        return "streaming-protocol-config.xml";
+    }
+
+    @Test
+    public void testStreamingWithInputStreamPayload() throws Exception
+    {
+        streamingTestCase(new ByteArrayInputStream(TEST_MESSAGE.getBytes()), SINGLE_ITERATION);
+    }
+
+    @Test
+    public void testStreamingWithStringPayload() throws Exception
+    {
+        streamingTestCase(TEST_MESSAGE, SINGLE_ITERATION);
+    }
+
+    /**
+     * @param payload    to be sent in each request
+     * @param iterations amount of times that the send flow should be called
+     * @throws Exception
+     */
+    private void streamingTestCase(Object payload, int iterations) throws Exception
+    {
+        final CountDownLatch latch = new CountDownLatch(1);
+        final AtomicReference<String> message = new AtomicReference<String>();
+        final AtomicInteger loopCount = new AtomicInteger(0);
+
+        EventCallback callback = new EventCallback()
+        {
+            @Override
+            public synchronized void eventReceived(MuleEventContext context, Object component)
+            {
+                try
+                {
+                    logger.info("called " + loopCount.incrementAndGet() + " times");
+                    FunctionalStreamingTestComponent ftc = (FunctionalStreamingTestComponent) component;
+                    // without this we may have problems with the many repeats
+                    if (1 == latch.getCount())
+                    {
+                        message.set(ftc.getSummary());
+                        assertEquals(format(RESULT, TEST_MESSAGE.length()), message.get());
+                        latch.countDown();
+                    }
+                }
+                catch (Exception e)
+                {
+                    logger.error(e.getMessage(), e);
+                }
+            }
+        };
+
+        // this works only if singleton set in descriptor
```

Comment:
This was a comment on the original transport streaming test.

It refers to this XML block in the test:
```
    <flow name="tcp-listen">
        <sockets:socket-listener config-ref="tcp-listener"/>
        <component>
            <!-- This component knows how to look for EOF -->
            <singleton-object class="org.mule.extension.socket.EOFStreamingTestComponent" />
        </component>
    </flow>
```